bonsai/ethereum-relay/src/uploader/pending_proofs/pending_proof_request_future.rs

```diff
@@ -98,22 +99,44 @@ impl Future for PendingProofRequest {
                     let response = futures::ready!(bonsai_get_receipt_fut.as_mut().poll(ctx));
                     match response {
                         Ok(receipt_response) => {
-                            if receipt_response.status == "SUCCEEDED" {
-                                return Poll::Ready(Ok(this.pending_proof_id.clone()));
+                            match (
+                                receipt_response.status.as_str(),
+                                receipt_response.receipt_url.is_some(),
+                            ) {
+                                ("SUCCEEDED", true) => {
+                                    return Poll::Ready(Ok(this.pending_proof_id.clone()))
+                                }
+                                ("SUCCEEDED", false) => {
```

Comment:
Yeah if this happens its a hard error and bonsai has something very wrong with it haha.